stdlib/format.mli

```diff
@@ -1071,10 +1071,20 @@ val asprintf : ('a, formatter, unit, string) format4 -> 'a
 val dprintf :
   ('a, formatter, unit, formatter -> unit) format4 -> 'a
 (** Same as {!fprintf}, except the formatter is the last argument.
-  All printing actions are delayed until the formatter is provided.
-  In particular, [printff "..." a b c] is a
-  function of type [formatter -> unit] which can be given to
-  a format specifier [%t].
+  In particular, [dprintf "..." a b c] is a function of type
+  [formatter -> unit] which can be given to a format specifier [%t].
+
+  This can be used as a replacement of {!asprintf} to delay formatting
+  decisions.  By using the string returned by {!asprintf} in a
+  formatting context, formatting decisions are taken in isolation and
+  the final string is created prematurely.  {!dprintf} allows to delay
+  such formatting decisions until the final formatting context is
+  known. For example:
```

Comment:
Perhaps: "Using the string returned by {!asprintf} in a
  formatting context forces formatting decisions to be taken in isolation, and
  the final string may be created prematurely.  {!dprintf} allows delay of
  such formatting decisions until the final formatting context is
  known."